util/conditions/merge_strategies.go

```diff
@@ -46,6 +46,9 @@ func WithConditionOrder(t ...clusterv1.ConditionType) MergeOption {
 
 // WithStepCounter instructs merge to add a "x of y completed" string to the message,
 // where x is the number of conditions with Status=true and y is the number passed to this method.
+//
+// NOTE: In order to avoid confusing message e.g 3 of 2, if the to value is lower than the actual
+// number of conditions, the to value is automatically increased up to the number of conditions.
 func WithStepCounter(to int) MergeOption {
```

Comment:
Can we get rid of the counter if we can count internally?

Reply:
The problem is that not all the conditions are created at the first reconciliation, so I have found it convenient to set the expected (minimum) number of conditions upfront
e.g. DockerMachine
containerCreating also, but not yet with a Bootstrapping condition --> steps 1 of 2, we know upfront the Bootstrapping should appear soon or later

Reply:
Even if the conditions aren't created at first reconciliation, we can greatly simplify the interface by inspecting the ones that are there and expand the counter on demand?

Reply:
fixed